eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php

```diff
@@ -1052,6 +1052,55 @@ public function testAssignSection()
     }
 
     /**
+     * Test search engine is updated after removal of the translation from all the Versions.
+     */
+    public function testRemoveTranslation()
+    {
+        $repository = $this->getRepository();
+        $searchService = $repository->getSearchService();
+        $contentService = $repository->getContentService();
+
+        $content = $this->createMultiLanguageContent(
+            [
+                'eng-US' => 'AmE Name',
+                'eng-GB' => 'BrE Name',
+            ],
+            2,
+            false
+        );
+
+        $contentService->removeTranslation($content->contentInfo, 'eng-GB');
+
+        $this->refreshSearch($repository);
+
+        // Test ContentId search returns Content without removed Translation
+        $query = new Query([
+            'query' => new Criterion\ContentId($content->contentInfo->id),
+            'filter' => new Criterion\LanguageCode('eng-GB', false),
+        ]);
+        $result = $searchService->findContent($query);
+        self::assertEquals(0, $result->totalCount);
+
+        // Test FullText search for removed unique name part returns no results
+        $query = new Query([
+            'query' => new Criterion\FullText('BrE'),
+        ]);
+        $result = $searchService->findContent($query);
+        self::assertEquals(0, $result->totalCount);
+
+        if (!$this->getSetupFactory() instanceof LegacyElasticsearchSetupFactory) {
+            // Test Location Search returns Content without removed Translation
+            $query = new LocationQuery(
+                [
+                    'query' => new Criterion\FullText('BrE'),
```

Comment:
Side question for @andrerom and @pspanja:
I had to wrap this in `if`, because apparently our ES implementation does not have Location FullText Criterion visitor (there's one for Content). Is it one of the things *todo* for ES or is it that this is just not possible for ES? (For Solr we have two separate FullText Criterion visitors).

If it's the latter case, then maybe [ES dispatcher](https://github.com/ezsystems/ezpublish-kernel/blob/v6.10.1/eZ/Publish/Core/Search/Elasticsearch/Content/CriterionVisitorDispatcher.php#L94-L96) should not throw exception? (it really doesn't solve the need for this if, but I still think it would be better - we also did something similar for Facets AFAIR)